tests/garage/experiment/test_resume.py

```diff
@@ -36,18 +35,17 @@ def fixture_exp():
 
 class TestResume(TfGraphTestCase):
     def test_resume(self):
-        # Manually create and remove temp folder
-        # Otherwise, tempfile unexpectedly removes folder in child folder
-        folder = tempfile.mkdtemp()
-        snapshotter.snapshot_dir = folder
-        snapshotter.snapshot_mode = 'last'
+        temp_dir = tempfile.TemporaryDirectory()
+
+        snapshot_config = SnapshotConfig(
+            snapshot_dir=temp_dir.name, snapshot_mode='last', snapshot_gap=1)
+        policy_params = fixture_exp(snapshot_config)
 
-        policy_params = fixture_exp()
         self.teardown_method()
```

Comment:
please make these separate test cases, rather than calling teardown/setup manually .that's an abuse of the test framework.